test/Sema/placeholder_type.swift

```diff
@@ -216,3 +216,16 @@ _ = (1...10)
     .map { (intValue, x: (_, boolValue: _)) in
         x.boolValue ? intValue : 0
     }
+
+// SR-15192
+
+let _: Array = [2]
+let _: [_] = [2]
+let _: Array<_> = [2]
+
+let _: Set = [2]
+let _: Set<_> = [2]
+
+let _: Dictionary = [2: "hi"]
+let _: [_: _] = [2: "hi"]
+let _: Dictionary<_, _> = [2: "hi"]
```

Comment:
Multiple ways to test this better:

- Add a SILGen test and very that the types are inferred using FileCheck
- Add a test that prints `-dump-ast` 
- Add a unit test to unittests/Sema that would construct a constraint system with placeholder contextual type and check solutions.

I'd prefer the last option but all of them a viable.